### PR TITLE
[CFP-2809] chore: update dependencies

### PR DESCRIPTION
**Pull request by gsamaniego41**
[CFP-2809](https://resolvn.atlassian.net/browse/CFP-2809)

### Summary
This PR updates the Material-UI dependency to version 7.

#### Datagrid updates
In MUI v6, the `valueGetter` and `valueFormatter` callback for a `GridColDef` received an object parameter containing multiple properties.

Example in v6:
```typescript
valueGetter: params => `${params.row.firstName} ${params.row.lastName}`,
```

In MUI v7, the signatures of `valueGetter` and `valueFormatter` were changed. Instead of passing a single object parameter, the callback now receives multiple arguments.

Example in v7:
```typescript
valueGetter: (_value, row) => `${row.firstName} ${row.lastName}`,
```

#### TreeView Deprecated
The component `TreeView` was deprecated and replaced by `SimpleTreeView`.

[CFP-2809]: https://resolvn.atlassian.net/browse/CFP-2809?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/DataGrid/components/DataGridCell.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { useGridApiContext, useGridRootProps } from '@mui/x-data-grid-premium'
 
 export const DataGridCell = React.forwardRef<HTMLDivElement, GridCellProps>(
   (props, ref) => {
-    const { colIndex, rowId, field, width, height } = props
```
Comment (jakeaustinlewis) [inline packages/components/src/DataGrid/components/DataGridCell.tsx:9]: `height` no longer exists as a property in the `GridCellProps` type

File: packages/components/src/SimpleTreeView/SimpleTreeView.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * - [Tree view](https://simspace.gitlab.io/engineering/ux-engineering/monorail/main/storybook/?path=/docs/data-display-treeview--default)
+ * - [Tree view (MUI)](https://mui.com/material-ui/react-tree-view/)
+ *
+ * API:
+ *
+ * - [SimpleTreeView API](https://mui.com/x/api/tree-view/simple-tree-view/)
+ */
+export const SimpleTreeView: typeof MuiTreeView = MuiTreeView
+
+export * from '@mui/x-tree-view/SimpleTreeView'
```
Comment (jakeaustinlewis) [inline packages/components/src/SimpleTreeView/SimpleTreeView.tsx:16]: The component [TreeView](https://mui.com/x/api/tree-view/tree-view/) has been deprecated and replaced with [SimpleTreeView]((https://mui.com/x/api/tree-view/simple-tree-view/)

File: packages/storybook/src/DataGrid/DataGridComponents.stories.tsx [hunk 3]
```diff
@@ -275,8 +275,8 @@ export const ToolbarGrid = story<DataGridProps>(args => {
       <DataGrid
         {...args}
         {...data}
-        components={{
```
Comment (jakeaustinlewis) [inline packages/storybook/src/DataGrid/DataGridComponents.stories.tsx:278]: `components` has been deprecated

File: packages/storybook/src/DataGrid/DataGridLayout.stories.tsx [hunk 4]
```diff
@@ -190,12 +190,12 @@ export const ColumnAlignment = story<DataGridProps<{}>>(
         },
       },
       {
-        field: 'fullName',
+        field: 'firstName',
```
Comment (gsamaniego41) [inline packages/storybook/src/DataGrid/DataGridLayout.stories.tsx:193]: `fullName`

File: packages/themes/src/meteor/components/DataGrid/themeOverrides.tsx [hunk 5]
```diff
@@ -127,9 +127,6 @@ export const MonorailDataGridOverrides: Components<Theme>['MuiDataGrid'] = {
         },
       }
     },
-    columnHeadersInner: {
-      height: '100%',
-    },
     columnHeaderDraggableContainer: () => {
```
[the remark this answers is not in the document]
Comment (gsamaniego41) [reply packages/themes/src/meteor/components/DataGrid/themeOverrides.tsx:130]: Current:
image

MUI X 7 update:
image

**Review by 0x706b — APPROVED** [review]

**Pull request merged**